proto/ptf/base_test.py

```diff
@@ -63,20 +61,19 @@ def stringify(n, length):
     'length' bytes, it is represented in the fewest number of bytes it
     does fit into without loss of precision.  It always returns a
     string at least one byte long, even if value=width=0."""
-    h = '%x' % n
-    s = ('0'*(len(h) % 2) + h).zfill(length*2).decode('hex')
-    return s
+    return int.to_bytes(n, length=length, byteorder='big')
 
 def ipv4_to_binary(addr):
     """Take an argument 'addr' containing an IPv4 address written as a
     string in dotted decimal notation, e.g. '10.1.2.3', and convert it
     to a string with binary contents expected by the Python P4Runtime
     client operations."""
     bytes_ = [int(b, 10) for b in addr.split('.')]
-    assert len(bytes_) == 4
-    # Note: The chr(b) call below will throw exception if any b is
-    # outside of the range [0, 255]], so no need to add a separate
-    # check for that here.
+    if len(bytes_) != 4:
+        raise ValueError("Invalid IPv4 address format")
+    for b in bytes_:
+        if b < 0 or b > 255:
+            raise ValueError("IPv4 address contains out-of-range value")
     return "".join(chr(b) for b in bytes_)
```

Comment:
this returns a `str`, while `stringify` returns a `bytes` object
for correctness and consistency, I believe this should also return a `bytes` object, which can easily be done with `bytes(bytes_)`, since the `bytes` constructor can take as input an array of integers

same for `mac_to_binary` below